File: commons/test-utils/src/main/java/com/cedarsoft/test/utils/OnlyLinux.java
package com.cedarsoft.test.utils;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.junit.jupiter.api.extension.*;

/**
 * Run tests only on Linux
 *
 */
@ExtendWith(DisableIfNotLinuxCondition.class)
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface OnlyLinux {
}

File: commons/test-utils/src/main/kotlin/com/cedarsoft/test/utils/OnlyLinux.kt
package com.cedarsoft.test.utils

import com.cedarsoft.test.utils.DisableIfNotLinuxCondition
import org.junit.jupiter.api.extension.ExtendWith

/**
 * Run tests only on Linux
 *
 */
@ExtendWith(DisableIfNotLinuxCondition::class)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class OnlyLinux
